src/ConfigurationClientWrapper.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AppConfigurationClient } from "@azure/app-configuration";
+
+const MaxBackoffDuration = 10 * 60 * 1000; // 10 minutes in milliseconds
+const MinBackoffDuration = 30 * 1000; // 30 seconds in milliseconds
+const MAX_SAFE_EXPONENTIAL = 30; // Used to avoid overflow. bitwise operations in JavaScript are limited to 32 bits. It overflows at 2^31 - 1.
+const JITTER_RATIO = 0.25;
+
+export class ConfigurationClientWrapper {
+    endpoint: string;
+    client: AppConfigurationClient;
+    backoffEndTime: number = 0; // Timestamp
+    failedAttempts: number = 0;
+
+    constructor(endpoint: string, client: AppConfigurationClient) {
+        this.endpoint = endpoint;
+        this.client = client;
+    }
+}
+
+export function updateClientBackoffStatus(clientWrapper: ConfigurationClientWrapper, successfull: boolean) {
```

Comment:
It is more appropriate to let this function be a public method of `ConfigurationClientWrapper` class.